library/core/src/lib.rs

```diff
@@ -230,6 +230,7 @@
 #![feature(unboxed_closures)]
 #![feature(unsized_fn_params)]
 #![feature(with_negative_coherence)]
+#![feature(proc_macro_hygiene)]
```

Comment:
why do we need this? I would prefer not enabling any new feature in the standard library

Reply:
It is required to use macro for loops.

Reply:
Error I got without the feature suggests to include it.
```
error[E0658]: custom attributes cannot be applied to expressions
  --> simple_while_loop_not_enabled.rs:15:5
   |
15 |     #[kani::loop_invariant(x >= 2)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #54727 <https://github.com/rust-lang/rust/issues/54727> for more information
   = help: add `#![feature(proc_macro_hygiene)]` to the crate attributes to enable
   = note: this compiler was built on 2024-11-02; consider upgrading it if it is out of date
```

Reply:
Yes, in that case, I will suggest that we go ahead and merge this as is, and fix this in Kani later. Can you change this to conditionally add it only if kani is enabled? I.e.:
```suggestion
#![cfg_attr(kani, feature(proc_macro_hygiene))]
```

Reply:
It works for Kani proof but will fail the Rust Tests.